c_src/main.c

```diff
@@ -78,17 +78,17 @@ erlfdb_future_cb(FDBFuture* fdb_future, void* data)
     ERL_NIF_TERM msg;
     bool cancelled;
 
-
-    enif_mutex_lock(future->lock);
+    enif_mutex_lock(future->cancel_lock);
     cancelled = future->cancelled;
-    enif_mutex_unlock(future->lock);
+    enif_mutex_unlock(future->cancel_lock);
```

Comment:
if we want to prevent duplicated message sending below, we should take the locker longer.